docs: fixed broken links and adopted htmltest - fixes #3834

themr0c opened the pull request:
docs: fixed broken links and adopted htmltest
chore: use the standard `httpd` container image to serve the docs locally

fixes #3834

```
make build_docs docs_check_links
podman run -v /home/ffloreth/src/gh/themr0c/crc:/antora:Z --rm docker.io/antora/antora:latest antora-playbook.yml
podman run -v /home/ffloreth/src/gh/themr0c/crc:/test:Z --rm docker.io/wjdp/htmltest:latest -c .htmltest.yml
htmltest started at 01:15:50 on docs/build
========================================================================
✔✔✔ passed in 1.345655164s
tested 10 documents
```

themr0c commented:
/retest

praveenkumar commented on antora-playbook.yml:
```diff
@@ -1,69 +1,68 @@
+    odo-docs-url: https://odo.dev/docs/introduction
+    odo-docs-url-installing: https://odo.dev/docs/overview/installation
+    telemetry-notice-url: https://developers.redhat.com/article/tool-data-collection
+    rhel-resolved-docs: https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/configuring_and_managing_networking/using-different-dns-servers-for-different-domains_configuring-and-managing-networking
```
These space related changes shouldn't be part of the commit.

themr0c commented:
Rebased and squashed.

themr0c committed:
docs: fixed broken links and adopted htmltest - fixes #3834

chore: keep the possibility to set the container with a variable

chore: serve docs with the default httpd container

Signed-off-by: Fabrice Flore-Thébault <ffloreth@redhat.com>

themr0c commented:
Fixed the features lost in rebase.
